bundler/lib/bundler/lazy_specification.rb

```diff
@@ -136,8 +136,12 @@ def __materialize__(candidates, fallback_to_non_installable: Bundler.frozen_bund
       end
       if search.nil? && fallback_to_non_installable
         search = candidates.last
-      else
-        search.dependencies = dependencies if search && search.full_name == full_name && (search.is_a?(RemoteSpecification) || search.is_a?(EndpointSpecification))
+      elsif search
```

Comment:
are we losing anything in the case where search is nil and `!fallback_to_non_installable` ?

Reply:
If `search` is `nil` and we don't want to fallback to non installable specifications, I think we end up returning `nil` which I believe is what we want?

Reply:
I think this is fine, but let me know if something does not square.